core/web/authentication.go

```diff
@@ -0,0 +1,138 @@
+package web
+
+import (
+	"chainlink/core/store"
+	"chainlink/core/store/models"
+	"chainlink/core/store/orm"
+	"net/http"
+
+	"github.com/gin-gonic/contrib/sessions"
+	"github.com/gin-gonic/gin"
+	"github.com/pkg/errors"
+)
+
+const (
+	// ExternalInitiatorAccessKeyHeader is the header name for the access key
+	// used by external initiators to authenticate
+	ExternalInitiatorAccessKeyHeader = "X-Chainlink-EA-AccessKey"
+	// ExternalInitiatorSecretHeader is the header name for the secret used by
+	// external initiators to authenticate
+	ExternalInitiatorSecretHeader = "X-Chainlink-EA-Secret"
+	// APIKey is the header name for the API token identifier for user authentication.
+	APIKey = "X-API-KEY"
+	// APISecret is the header name for the API token secret for user authentication.
+	APISecret = "X-API-SECRET"
+)
+
+var (
+	// ErrorAuthFailed is a generic authentication failed - but not because of
+	// some system failure on our behalf (i.e. HTTP 5xx), more detail is not
+	// given
+	ErrorAuthFailed = errors.New("Authentication failed")
+)
+
+type authType func(store *store.Store, ctx *gin.Context) error
+
+func authenticatedUser(c *gin.Context) (*models.User, bool) {
+	obj, ok := c.Get(SessionUserKey)
+	if !ok {
+		return nil, false
+	}
+	return obj.(*models.User), ok
+}
+
+func eiAuth(store *store.Store, c *gin.Context) error {
```

Comment:
Suggestion: Would rather not abbreviate it to `eiAuth`, prefer the method name to be a verb: `authenticateExternalInitiator`. I did prefer the previous name of `sessionAuth` since this logic works with sessions (`c.Set(SessionExternalInitiatorKey, ei)`) that's not immediately obvious anymore.

Reply:
The sessionAuth is for user sessions not external initiators

Reply:
Addressed in
https://github.com/smartcontractkit/chainlink/pull/1949/commits/81682b4bec94eb8e29f97d16662bd6a515b7ab1e